packages/ts-transform/src/utils/css-transform.tsx

```diff
@@ -19,33 +20,42 @@ const minify = () => {
     });
 };
 
-const COMMA_CHAR_CODE = 58;
+const removeSpacesBeforeSelector = (selector: parser.Node) => (selector.spaces.before = '');
+const doesCombinatorTypePrecedesSelector = (selector: parser.Node) => {
+  const previousSelector = selector.prev();
 
+  return previousSelector && previousSelector.type === 'combinator';
+};
+const prependNestingTypeToSelector = (selector: parser.Node) => {
+  const { parent } = selector;
+
+  if (parent) {
+    const nesting = parser.nesting({});
+
+    parent.insertBefore(selector, nesting);
+  }
+};
 const parentOrphenedPseudos = plugin('parent-orphened-pseudos', () => {
   return (root) => {
     root.walkRules((rule) => {
-      if (rule.selector.includes(':')) {
-        const newSelector = rule.selector
-          .replace(/\s+/g, ' ')
-          .split(', ')
-          .map((part) => {
-            if (part.match(/^. /)) {
-              // If the selector has one characters with a space after it, e.g. "> :first-child" then return early.
-              return part;
-            }
+      const parserRoot = parser((selectors) => {
+        selectors.walk((selector) => {
+          removeSpacesBeforeSelector(selector);
```

Comment:
nit: if we don't need this to make our code work we can probably just remove this? i had it so it made the silly string splitting easier 😄

Reply:
hmm..If I remove this, tests got failed because of leading spaces with following error:

<img width="804" alt="Screen Shot 2020-05-31 at 4 46 22 PM" src="https://user-images.githubusercontent.com/7638708/83351036-48b07a80-a35e-11ea-8f6d-91f0b74de154.png">

Reply:
thats ok just update snapshots

Reply:
@madou This looks bad if don't remove spaces before pseudo selectors:

<img width="604" alt="Screen Shot 2020-05-31 at 9 30 49 PM" src="https://user-images.githubusercontent.com/7638708/83356814-26325780-a386-11ea-9e7a-cbcd752d9969.png">

Reply:
! no it isn't 🤔 

let's leave it as is for now 😄 

fyi emotions implementation of this is here https://github.com/emotion-js/emotion/blob/5e39d81bf9177e68e78d23bd869c1592341d9768/packages/cache/src/stylis-plugins.js#L49 if we need some inspiration

Reply:
There is a `lossless` settings under [ProcessorOptions](https://github.com/postcss/postcss-selector-parser/blob/master/API.md#processoroptions). If we set it to false, things works as expected. No need to manually remove the spaces 🤘 

hmm..Emotion one is bit different it seems..anyways `lossless` solves our problem 😃